amavisd-milter/mlfi.c

```diff
@@ -331,20 +338,23 @@ mlfi_connect(SMFICTX *ctx, char *hostname, _SOCK_ADDR * hostaddr)
     mlfi->mlfi_amasd = -1;
 
     /* Save connection informations */
-    if ((client_name = smfi_getsymval(ctx, "{client_name}")) != NULL) {
-	logmsg(LOG_INFO, "%s: Client name: %s", hostname, client_name);
-	if ((mlfi->mlfi_client_host = strdup(client_name)) == NULL) {
-	    logmsg(LOG_ERR, "%s: could not allocate memory", hostname);
-	    mlfi_setreply_tempfail(ctx);
-	    return SMFIS_TEMPFAIL;
-	}
+    if ((mlfi->mlfi_client_host = strdup(client_name)) == NULL) {
```

Comment:
This should be `strdup(client_host)` instead of `client_name`. _name is not defined yet.

Reply:
You are absolutely right.